File: WRUW/GroupedProgramsTableViewController.m
#import "GroupedProgramsTableViewController.h"
#import "ARAnalytics.h"

@interface GroupedProgramsTableViewController ()
{
    NSMutableArray *daysOfWeek;
}

@end

@implementation GroupedProgramsTableViewController

-(void) prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender{
    
    if ([segue.identifier isEqualToString:@"programsSegue"]) {
        ShowsTableViewController *stvc = [segue destinationViewController];
        NSIndexPath *path = [self.tableView indexPathForSelectedRow];
        
        int day = (int)path.row;
        [stvc setDayOfWeek:day];
    }
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self setTitle:@"Programs"];
    [ARAnalytics event:@"Screen view"
        withProperties:@{ @"screen": @"Programs View" }];

    daysOfWeek = [[NSMutableArray alloc] initWithObjects:@"Any", @"Sunday", @"Monday", @"Tuesday", @"Wednesday", @"Thursday", @"Friday", @"Saturday", nil];
}

#pragma mark - Table view data source

- (NSString *)tableView:(UITableView *)tableView
titleForHeaderInSection:(NSInteger)section {
    return @"Pick A Day";
}

- (NSInteger)tableView:(UITableView *)tableView
 numberOfRowsInSection:(NSInteger)section {
    // Return the number of rows in the section.
    return daysOfWeek.count;
}


- (UITableViewCell *)tableView:(UITableView *)tableView
         cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    static NSString *CellIdentifier = @"Cell";

    UITableViewCell *cell =
        [tableView dequeueReusableCellWithIdentifier:CellIdentifier];

    if (cell == nil) {
        cell =
            [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault
                                   reuseIdentifier:CellIdentifier];
        cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
    }
    cell.textLabel.text = [daysOfWeek objectAtIndex:indexPath.row];
    
    return cell;
}

@end

File: WRUW/View Controllers/GroupedProgramsTableViewController.swift
import ARAnalytics
import UIKit

class GroupedProgramsTableViewController: UITableViewController {
    private lazy var daysOfWeek = ["Any", "Sunday", "Monday", "Tuesday",
                                   "Wednesday", "Thursday", "Friday",
                                   "Saturday"]
    private let cellIdentifier = "Cell"

    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if let showsViewController = segue.destination as? ShowsTableViewController,
            let day = tableView.indexPathForSelectedRow?.row {

            showsViewController.dayOfWeek = Int32(day)
        }
    }

    override func viewDidLoad() {
        super.viewDidLoad()

        title = "Programs"
        ARAnalytics.event("Screen view",
                          withProperties: ["screen": "Programs View"])
    }

    override func tableView(_ tableView: UITableView,
                            titleForHeaderInSection section: Int) -> String? {
        return "Pick A Day"
    }

    override func tableView(_ tableView: UITableView,
                            numberOfRowsInSection section: Int) -> Int {
        return daysOfWeek.count
    }

    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell =
            tableView.dequeueReusableCell(withIdentifier: cellIdentifier) ??
            UITableViewCell(style: .default, reuseIdentifier: cellIdentifier)

        cell.accessoryType = .disclosureIndicator

        cell.textLabel?.text = daysOfWeek[indexPath.row]

        return cell
    }
}
